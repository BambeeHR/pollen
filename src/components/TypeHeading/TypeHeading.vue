<template functional>
  <component
    :is="props.tag"
    :class="[
      `type-heading-${props.variant}`,
      `type-heading-${props.variant}__${props.weight}`,
      data.class,
      data.staticClass,
    ]"
    class="type-heading"
    v-bind="$options.getAttributes({ data, parent })"
    v-on="listeners"
  >
    <slot />
  </component>
</template>

<script>
import Typography from '../../constants/Typography';
import getAttributes from '../../util/getAttributes';

const { Heading, Weights } = Typography;

export default {
  props: {
    tag: {
      type: String,
      default: 'span',
    },
    variant: {
      type: String,
      default: Heading.HEADLINE,
      validator: (value) => Object.values(Heading).includes(value),
    },
    weight: {
      type: String,
      default: Weights.STRONG,
      validator: (value) => [Weights.REGULAR, Weights.STRONG].includes(value),
    },
  },
  getAttributes,
};
</script>

<style>
/* Intentionally left unscoped to provide global helper styles. */
@each $variant in subtitle, headline {
  .type-heading-$(variant) {
    @apply font-heading-$(variant);
  }

  .type-heading-$(variant)__regular {
    @apply font-normal;
  }

  .type-heading-$(variant)__strong {
    @apply font-semibold;
  }
}

.type-heading-title__regular {
  @apply font-heading-title;
}

.type-heading-title__strong {
  @apply font-heading-title-special;
}
</style>
