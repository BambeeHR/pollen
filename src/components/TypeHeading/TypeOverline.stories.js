import Typography from '../../constants/Typography';
import TypeHeading from './TypeHeading.vue';
import TypeOverline from '../TypeOverline/TypeOverline.vue';

const { Heading, Weights } = Typography;

export default {
  title: 'TypeHeading',
  component: TypeHeading,
};

export const Gallery = () => ({
  components: { TypeHeading, TypeOverline },
  data() {
    return {
      variants: Object.values(Heading),
      weights: [Weights.REGULAR, Weights.STRONG],
    };
  },
  template: `
    <div>
      <div v-for="variant in variants" :key="variant" :style="{ marginBottom: '48px' }">
        <TypeOverline tag="h1">{{ variant }}</TypeOverline>
        <TypeHeading 
          v-for="weight in weights"
          :key="variant + '-' + weight"
          :variant="variant"
          :weight="weight"
          tag="div"
        >
          The quick brown fox jumps over the lazy dog.
        </TypeHeading>
      </div>
    </div>
  `,
});
