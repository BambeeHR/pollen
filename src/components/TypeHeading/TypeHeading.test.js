import { mount } from '@vue/test-utils';
import TypeHeading from './TypeHeading.vue';

describe('TypeHeading', () => {
  test('renders correctly', () => {
    const wrapper = mount(TypeHeading, {
      slots: {
        default: 'Hello world!',
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
