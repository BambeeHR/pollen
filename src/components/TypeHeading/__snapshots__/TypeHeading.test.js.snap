// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TypeHeading renders correctly 1`] = `
<span
  class="type-heading type-heading-headline type-heading-headline--strong"
>
  Hello world!
</span>
`;
